packages/flame/lib/src/geometry/circle_component.dart

```diff
@@ -83,6 +83,11 @@ class CircleComponent extends ShapeComponent {
         scaledRadius * scaledRadius;
   }
 
+  @override
+  bool containsLocalPoint(Vector2 point) {
+    return (point * 2 - size).length2 <= size.x * size.x;
```

Comment:
```suggestion
    return ((point * 2)..sub(size)..length2 <= size.x * size.x;
```
To avoid creating two new `Vector2`s instead of one

Reply:
I've changed the logic to avoid creating any vectors at all.